### PR TITLE
mxrtc

### PR DESCRIPTION
**Pull request by piiswrong**

### CODE REVIEW

File: include/mxnet/c_api.h [hunk 1]
```diff
@@ -48,6 +48,8 @@ typedef void *DataIterHandle;
 typedef void *KVStoreHandle;
 /*! \brief handle to RecordIO */
 typedef void *RecordIOHandle;
+/*! \brief handle to MXRtc*/
+typedef void *RtcHandle;
```
Comment (tqchen) [inline include/mxnet/c_api.h]: consider rename Rtc to RTC? as it is abbrevation for runtime compile

File: include/mxnet/mxrtc.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    static const std::string str_type;
+    static std::unordered_map<std::string, char*> kernel_registry;
+
+    std::string name_;
+    index_t num_input_, num_output_;
+    std::string code_;
+    char* ptx_;
+    std::unordered_map<int, CUmodule> module_;
+    std::unordered_map<int, CUfunction> func_;
+
```
Comment (tqchen) [inline include/mxnet/mxrtc.h]: Use google C style for function naming

**Comment by vchuravy** [comment]
This should probably be rebased to get rid of the superfluous commits. Otherwise this looks exciting ;)

**Comment by tqchen** [comment]
Thanks for exciting improvements!

As this touches the include folder, which is user facing, everything need to documented in detail with doxygen format and other headers, please do this soon in another PR, also conform the code style to Google's C style, e.g.

- CamelCase function name

**Pull request merged**